### PR TITLE
Fixes concurrent modification exception in ColumnIndices class

### LINKED ISSUE #6876: Update 6.1.0 to 7.0.0 - ConcurrentModificationException crashs
Hi,

Since I have updated to realm 7.0.0 (previoulsly 6.1.0), some ConcurrentModificationException crashs happen.
I don't see any code changes that could explain this crash, except for the realm SDK update.

Unlikely, I have never reproduct this issue myself. I get this exception by crashlytics. So I can't give you the reproduct steps.

As you have a better knowledge of the Realm SDK than me, I expect you to help me find out the source of this issue.

The error indicates a concurrent modification, but can you tell me if there are some kinds of modifications that have more probability to provoke this crash?

#### Version of Realm and tooling
Realm version: 7.0.0
Realm Sync feature enabled: No
Android Studio version: 3.6.3
Android Build Tools version: 29.0.3
Gradle version: 6.4.1

Trace :+1: Fatal Exception: java.util.ConcurrentModificationException
       at java.util.HashMap$HashIterator.nextNode(HashMap.java:1441)
       at java.util.HashMap$EntryIterator.next(HashMap.java:1475)
       at java.util.HashMap$EntryIterator.next(HashMap.java:1473)
       at io.realm.internal.ColumnIndices.refresh(ColumnIndices.java:121)
       at io.realm.RealmSchema.refresh(RealmSchema.java:260)
       at io.realm.BaseRealm$1.onSchemaChanged(BaseRealm.java:97)
       at io.realm.internal.Table.nativeFreeze(Table.java)
       at io.realm.internal.Table.freeze(Table.java:676)
       at io.realm.internal.OsResults.freeze(OsResults.java:324)
       at io.realm.RealmResults.freeze(RealmResults.java:542)
       at io.realm.rx.RealmObservableFactory$6$1.onChange(RealmObservableFactory.java:196)
       at io.realm.rx.RealmObservableFactory$6$1.onChange(RealmObservableFactory.java:192)
       at io.realm.internal.ObservableCollection$RealmChangeListenerWrapper.onChange(ObservableCollection.java:39)
       at io.realm.internal.ObservableCollection$CollectionObserverPair.onChange(ObservableCollection.java:20)
       at io.realm.internal.ObservableCollection$Callback.onCalled(ObservableCollection.java:64)
       at io.realm.internal.ObservableCollection$Callback.onCalled(ObservableCollection.java:54)
       at io.realm.internal.ObserverPairList.foreach(ObserverPairList.java:109)
       at io.realm.internal.OsResults.notifyChangeListeners(OsResults.java:638)
       at android.os.MessageQueue.nativePollOnce(MessageQueue.java)
       at android.os.MessageQueue.next(MessageQueue.java:326)
       at android.os.Looper.loop(Looper.java:181)
       at android.os.HandlerThread.run(HandlerThread.java:65)

### PR DESCRIPTION
**Pull request by clementetb**
ColumnIndices was not thread-safe and could throw concurrent modification exceptions if the app tried, at the same time, to refresh a schema and to get info about a model.

closes #6876

### CODE REVIEW

**Review by rorbech — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,18 @@
+## 7.0.6 (yyyy-mm-dd)
+
+### Enhancements
+* None.
+
+### Fixes
+* Fixes concurrent modification exceptions when refreshing a Realm (Issue [#6876](https://github.com/realm/realm-java/issues/6876))
+
+### Compatibility
+* None.
```
Comment (cmelchior) [inline CHANGELOG.md:10]: This is not correct. You can just copy the compatibility section from 7.0.5

**Review by cmelchior — APPROVED** [review]
Nice 👏

**Review by edualonso — APPROVED** [review]
Nice catch.

File: CHANGELOG.md (hunk 1 above)
Comment (cmelchior) [reply CHANGELOG.md]: I copied the section

**Pull request merged**